Analysis/CodeAnalyzer.cs

```diff
@@ -81,16 +85,12 @@ private void SimplifiedAnalysis()
         /// <param name="depth">Current depth</param>
         private void ASTreeDFS(INode node, int depth)
         {
+            foreach (var smell in AstSmells) smell.AnalyzeNode(node, depth);
 
-            foreach (var smell in AstSmells)
-            {
-                smell.AnalyzeNode(node, depth);
-            }
-
-            foreach (var child in node.ChildNodes)
-            {
-                ASTreeDFS(child, depth + 1);
-            }
+            //foreach (var child in node.ChildNodes)
+            //{
+            //    ASTreeDFS(child, depth + 1);
+            //}
         }
```

Comment:
Unfortunately you are moving the logic of depth first traversal of the AST to individual smells which results in every single one of them going through the entire tree on it's own.

Reply:
I've just fixed it.